pkg/apis/metalkube/v1alpha1/baremetalhost_types.go

```diff
@@ -165,6 +165,16 @@ type BareMetalHost struct {
 	Status BareMetalHostStatus `json:"status,omitempty"`
 }
 
+// Available returns true if the host is available to be provisioned.
+func (host *BareMetalHost) Available() (bool, error) {
```

Comment:
What kinds of cases are you thinking might return an error? The response will be based on data in the status of the object, right?

Reply:
Yep, good point. We don't want outside code initiating any kind of active inspection, so I'll get rid of the error.